cmd/oras/cp.go

```diff
@@ -52,54 +49,61 @@ func copyCmd() *cobra.Command {
 
 ** This command is in preview and under development. **
 
-Example - Copy the artifact tagged with 'v1' from repository 'localhost:5000/net-monitor' to repository 'localhost:5000/net-monitor-copy' 
-  oras cp localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1
-
-Example - Copy the artifact tagged with 'v1' and its referrers from repository 'localhost:5000/net-monitor' to 'localhost:5000/net-monitor-copy'
-  oras cp -r localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1
-
-Example - Copy the artifact tagged with 'v1' from repository 'localhost:5000/net-monitor' to 'localhost:5000/net-monitor-copy' with certain platform
-  oras cp --platform linux/arm/v5 localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1 
-
-Example - Copy the artifact tagged with 'v1' from repository 'localhost:5000/net-monitor' to 'localhost:5000/net-monitor-copy' with multiple tags
-  oras cp localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1,tag2,tag3
-
-Example - Copy the artifact tagged with 'v1' from repository 'localhost:5000/net-monitor' to 'localhost:5000/net-monitor-copy' with multiple tags and concurrency level tuned
-  oras cp --concurrency 6 localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1,tag2,tag3
+Example - Copy the artifact tagged 'v1':
+  oras cp localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy:v1  # copy between repositories
+  oras cp localhost:5000/net-monitor:v1 localhost:5000/net-monitor-copy     # copy without tagging in the destination
```

Comment:
I think from a customers perspective `--to-oci` would be to an oci registry. I feel like it would be more intuitive to say `--to-file` or a little worse to me `--to-directory`

Reply:
Thinking about it more, I like the `--to=format` and `--from=format` better because that would leave the possibility of other formats other than registry and directory. Having a TGZ format would be very handy for example. Having an input of stdin or stndout might be convenient as well.

Reply:
The oci image layout is not a **format** but a **layout**, see [spec](https://github.com/opencontainers/image-spec/blob/main/image-layout.md)

So for short flags, how about `--from-layout`, `--to-layout` for `oras cp` and `--layout` to other commands?

`--layout` is equivalent to `--layout type=oci`. To support more layouts in the future, we can add `--from-layout type=docker` (just taking docker image layout as an example, doesn't mean we need to support it)

Reply:
Can you elaborate on the scenarios of using an input of stdin? To me the UX sounds looks like blew:

> some-cmd | oras cp - oci://localhost/owner/name:v1

Reply:
Also for uploading `.tgz` image layout, I might take a closer look tomorrow to see if it's already supported or what we need to do to support it.

Reply:
@TerryHowe The original design is `--to-target type=oci` and `--from-target type=oci`, which support more types / formats. For better UX, we choose `--to-oci` to be an alias of `--to-target type=oci`. The commit 83c4af14c04c1c438e7189422a0381201965dcde in this PR shows the original design.

Since we only have one type, it would be better hide the full form and introduce later.

@qweeah We need to document the original design somewhere so that later we can follow.

Reply:
> Having an input of stdin or stndout might be convenient as well.

Currently, `oras` / `oras-go` requires random access for read and write for performance considerations, and thus `stdin` and `stdout` are not supported. Maybe we can consider that after `oras v1.0.0`, trading convenience off performance.

Reply:
I don't want to get too side tracked by all the possible formats and destinations, but also don't want to design into a corner. It definitely might be convenient to support stdin/stdout for example, but maybe that should be done in another command. The tgz thing also might make more sense for another command since tgz input/output make sense for multiple repositories.

Reply:
> It definitely might be convenient to support stdin/stdout for example, but maybe that should be done in another command. The tgz thing also might make more sense for another command since tgz input/output make sense for multiple repositories.

@TerryHowe Can you help create enhancement issues with detailed scenario for those? We can continue discussion there.

Reply:
> I think from a customers perspective --to-oci would be to an oci registry. I feel like it would be more intuitive to say --to-file or a little worse to me --to-directory.

To avoid ambiguity, how about using `--to-oci-layout` and `--from-oci-layout` in `oras copy` respectively?

Reply:
https://github.com/oras-project/oras/issues/751